### DISCUSSION TITLE
Specify executable name

### DISCUSSION THREAD

**Discussion by thenoobtester** [pyinstaller/pyinstaller#7064] [Help] [answered]
Hi,

I'm trying to run [Mimtproxy](https://mitmproxy.org/) binaries on a system that doesn't have a recent libc (Debian 9 stretch), when I run into this type of issue I usually get around (with some tinkering obviously) by using a combo of LD_LIBRARY_PATH, LD_PRELOAD with the needed libc version and a more recent ld-linux-x86-64.so.2. I tried with mitmproxy executable which is built with pyInstaller however it uses ld-linux-x86-64.so.2 as the name of the executable and therefore fails to find a pyInstaller package to unpack and run, here is it's error message:

`[4990] Cannot open PyInstaller archive from executable (/mylibs/ld-linux-x86-64.so.2) or external archive (/mylibs/ld-linux-x86-64.so.2.pkg)`

And here is the command line I used to run it:

`LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/mylibs/ /mylibs/ld-linux-x86-64.so.2 ./mitmproxy`

It would help if there was a way to specify the name of the executable/package for pyInstaller to run. I'll gladly admit this is pretty messed up but this is the situation I'm in:)

PS:I went through pyi_main.c and didn't see anything that could help me.

Thank you.

**Comment by rokm** [comment, marked as answer]
Sounds like you would need to disable the executable path resolution via `/proc/self/exe` in the bootloader, by removing [this block](https://github.com/pyinstaller/pyinstaller/blob/a54d89e161c555837113dc399421f59111b8b654/bootloader/src/pyi_path.c#L297-L303).

But since you're going to need to rebuild the bootloader anyway, why not do so on Debian 9 stretch, and use PyInstaller there to create a binary of mitmproxy? That way, it's going to be compatible with the version of libc on your target system...

**Reply by thenoobtester** [reply]
Thank you for the feedback, it seems indeed as I would have to rebuild something, however I just didn't feel like making a binary of mitmproxy myself with PyInstaller as its creator described the process as : 

`(The whole process also is a giant PITA. If you can avoid it, avoid it. We hopefully reach the point where tools like PyOxidizer become mature enough to be usable, but that hasn't been the case yet)` Link: https://stackoverflow.com/questions/62334867/how-are-the-pre-compiled-mitmproxy-binaries-actually-built

Anyways, thank you for the response, I found an older version which works fine.
